### DISCUSSION TITLE
Rotating a in-built geometric objects

### DISCUSSION THREAD

**Discussion by raksharuia** [NanoComp/meep#2448] [Q&A] [answered]
Hello,

I started using MEEP form research a few months ago. Maybe this was discussed earlier but I could not find it in the forum. 

Is it possible to rotate geometries along any axis (defined) to create different geometries. for example, rotate a square to form a diamond. I have two superposing basic geometries (2 blocks) overlapping each other and I want to rotate one of them by a certain angle to construct some shapes. 

Are there any straight forward ways to define rotation in the geometry object/utilities?

**Comment by stevengj** [comment, marked as answer]
Currently, the objects don't have a `rotate` method but it would be easy to add one.  https://github.com/NanoComp/meep/issues/2449

For now, you can rotate them during the construction of the object.  e.g. you can create a rotated square by passing rotated `e1` and `e2` vectors to the `Block` constructor.  ([vectors](https://meep.readthedocs.io/en/latest/Python_User_Interface/#vector3) *do* have a `rotate` method.)

**Reply by raksharuia** [reply]
Thank you for replying @stevengj.  A follow-up question, simply to clarify. 

Assuming I am constructing 2 partially overlapping squares and I rotate one of them, during construction by rotating input vectors to the Block. Am I correct in assuming that it would only reflect on how the geometry/epsilon function? This should not affect the  simulations further on; for example the  run function of the Modesolver class while calculating bands?

**Reply by stevengj** [reply]
The geometry specifies the distribution of materials (ε).  Of course, the materials affects results later on (but they don't directly affect your *code* later on, if that's what you're asking).

**Reply by mprobst6** [reply]
If you pass a rotated `e1` and `e2` to the `Block` constructor and define the block to have an anisotropic medium, does the medium also get rotated so that the axes of the medium align with the axes of the `Block` (case 2) or do they maintain their original orientation (case 1)? Would I have to separately rotate the permittivity using the transform method if I wanted to keep the axes aligned?

```python
mp.Block(
    material = anisotropic_material,
    center=mp.Vector3(0,0,0),
    size=mp.Vector3(mp.inf,width,0),
    e1=mp.Vector3(x=1).rotate(mp.Vector3(z=1),rotation),
    e2=mp.Vector3(y=1).rotate(mp.Vector3(z=1),rotation)
)
```

![rotation](https://user-images.githubusercontent.com/89593456/226754434-8ddbf7e7-ee21-4af6-be7f-3c8a637211e2.png)
